lib/paginate.js

```diff
@@ -55,3 +55,31 @@ module.exports = {
     );
   },
 };
+
+/**
+ * Async generator to facilitate extremely large queries whose results
+ * cannot fit in memory all at once. The given query must accept an "offset"
+ * param that will be used to paginate the query.
+ *
+ * Should be used like this:
+ *
+ * for await (const row of paginateQueryAsync(sql, params)) {
+ *  // do something with row
+ * }
+ *
+ */
+module.exports.paginateQueryAsync = async function* (sql, params) {
```

Comment:
I'm experimenting with async generators for the first time, so extra eyes on this would be appreciated!